pyvrp/Population.py

```diff
@@ -140,21 +151,40 @@ def clear(self):
         self._feas = SubPopulation(self._op, self._params)
         self._infeas = SubPopulation(self._op, self._params)
 
-    def get_binary_tournament(self, rng: XorShift128) -> Individual:
+    def get_binary_tournament(
+        self,
+        rng: XorShift128,
+        cost_evaluator: CostEvaluator,
+        update_fitness: bool = True,
+    ) -> Individual:
         """
         Selects an individual from this population by binary tournament.
 
         Parameters
         ----------
         rng
             Random number generator.
+        cost_evaluator
+            Cost evaluator to use when computing the fitness.
+        update_fitness
+            Boolean whether to update the fitness values before getting the
+            binary tournament. Can be set to False for efficiency if it is
+            known that the fitness has not changed, e.g. when calling this
+            function repeatedly.
 
         Returns
         -------
         Individual
             The selected individual.
         """
 
+        if update_fitness:
+            # Note: even though the cost does not change, for the feasible
+            # subpopulation, we must always call compute_fitness as it is not
+            # updated when adding individuals
+            self._feas.update_fitness(cost_evaluator)
+            self._infeas.update_fitness(cost_evaluator)
```

Comment:
How much does it cost us if we do this every time, instead of having a flag that determines whether to update yes/no?

Reply:
Good idea, done